server/src/main/java/org/elasticsearch/gateway/GatewayMetaState.java

```diff
@@ -164,22 +176,55 @@ public void applyClusterState(ClusterChangedEvent event) {
         }
 
         if (event.state().blocks().disableStatePersistence()) {
-            // reset the current state, we need to start fresh...
-            previousMetaData = null;
-            previousManifest = null;
+            incrementalWrite = false;
             return;
         }
 
         try {
-            if (previousManifest == null) {
-                previousManifest = metaStateService.loadManifestOrEmpty();
-            }
-            updateMetaData(event);
-        } catch (Exception e) {
+            updateClusterState(event.state(), event.previousState());
+            incrementalWrite = true;
+        } catch (WriteStateException e) {
             logger.warn("Exception occurred when storing new meta data", e);
         }
     }
 
+    @Override
+    public long getCurrentTerm() {
+        return previousManifest.getCurrentTerm();
+    }
+
+    @Override
+    public ClusterState getLastAcceptedState() {
+        assert previousClusterState.nodes().getSize() > 0 : "Call setLocalNode before calling this method";
```

Comment:
why not check that `getLocalNode() != null`? It's a stronger check than what's currently being checked here.

Reply:
https://github.com/elastic/elasticsearch/pull/35819/commits/24c43fa135b603e1e2c5dca70182a80a3e05d35d